File: Problem3.js
/**
 * Created by sle7 on 10/2/2015.
 */

/*
    Largest Prime Factor
 */

var largestPrimeFactor = function(num) {
    var largest = 0;
    var curFactor = 2;
    var current = true;
    while(largest < num) {
        while(current) {
            if(num % curFactor === 0) {
                largest = curFactor;
                num = num/curFactor;
            }
            else {
                current = false;
            }
        }
        curFactor++;
        current = true;
    }
    return largest;
};

console.log(largestPrimeFactor(600851475143));

File: Problem3.rb
# Created by sle7 on 12/3/2015.

#
#   Largest Prime Factor
#

n = 600_851_475_143

def largestPrimeFactor(n)
  cur = 2
  largest = 0
  while largest < n
    if n % cur == 0
      largest = cur
      n /= cur
    end
    cur += 1
  end
  puts largest
end

largestPrimeFactor(n)
